src/toucan2/tools/identity_query.clj

```diff
@@ -74,6 +74,14 @@
 ;;;; Identity connection
 
 (deftype ^:no-doc IdentityConnection []
+  java.sql.Wrapper
+  (isWrapperFor [this klass]
+    (= klass java.sql.Connection))
+  (unwrap [this klass]
+    (if (.isWrapperFor this klass)
+      this
+      (throw (ex-info (str "IdentityConnection is not the wrapper for " klass) {}))))
+
```

Comment:
For whatever reason, `.unwrap` hits this object even though it doesn't implement `java.sql.Connection`, and thus `extend java.sql.Connection` should not apply to it. Anyway, to appease the weird behavior, I've implemented Wrapper for it. Implementing whole j.s.Connection would be an overkill.